src/aws-cpp-sdk-core/source/smithy/client/AwsSmithyClientBase.cpp

```diff
@@ -388,8 +388,15 @@ void AwsSmithyClientBase::HandleAsyncReply(std::shared_ptr<AwsSmithyClientAsyncR
             {
               assert(m_errorMarshaller);
               const Aws::String regionFromResponse = m_errorMarshaller->ExtractRegion(outcome.GetError());
-              const Aws::String& signerRegion = pRequestCtx->m_endpoint.GetAttributes()->authScheme.GetSigningRegion() ?
-                                                pRequestCtx->m_endpoint.GetAttributes()->authScheme.GetSigningRegion().value() : "";
+              const Aws::String signerRegion = [&]() {
+                  if (!regionFromResponse.empty() &&
+                      pRequestCtx->m_endpoint.GetAttributes() &&
```

Comment:
the problem here was that lack of the check for an Optional `pRequestCtx->m_endpoint.GetAttributes()`
which is not always set.